units.md

```diff
@@ -0,0 +1,82 @@
+> _NOTE:_ The manipulation of very large or very small numbers will be simplified by the use of [metric prefixes](https://en.wikipedia.org/wiki/Metric_prefix) of the [SI](https://en.wikipedia.org/wiki/International_System_of_Units).
+
+## Targeted Audience
+
+This specification is intended for developers who wish to interact with the Massa blockchain network using various tools and applications such as node API, smart contract tooling, dApps tooling, Thyra, and Thyra plugins.
+
+## Units
+
+### Massa coin
+
+Massa coin is the name of the native coin on the Massa blockchain.
+Massa coin is expressed as an unsigned 64-bit integer.
+
+1 Massa is equivalent to 1,000,000,000 nanoMassa, the smallest unit of Massa coin.
+
+To simplify usage, the following standard units implemenation are recommended:
+
+- milliMassa (mMASSA): 1 milliMassa represents 1,000,000 nanoMassa and also 1/1000th of a Massa.
+- microMassa (µMASSA): 1 microMassa represents 1,000 nanoMassa and also 1/1,000,000th of a Massa.
+- nanoMassa (nMASS): 1 nanoMassa represents the smallest unit of Massa coin and also 1/1,000,000,000th of a Massa.
+
+### Gaz
+
+Gaz is a virtual unit used to measure the computational complexity of executing transactions on the Massa blockchain. Each operation requires a certain amount of Gaz, and the amount of Gaz consumed is proportional to the computational complexity of the operation.
+Gaz is expressed as an unsigned 64-bit integer.
+
+1 Amphora is equivalent to 1,000,000,000 nanoAmphora, the smallest unit of gaz.
+
+To simplify usage, the following standard units implemenation are recommended:
+
+- milliAmphora (mAMPHORA): 1 milliAmphora represents 1,000,000 nanoAmphora or 1/1000th of an Amphora.
+- microAmphora (µAMPHORA): 1 microAmphora represents 1,000 nanoAmphora or 1/1,000,000th of an Amphora.
+- nanoAmphora (nAMPHORA): 1 nanoAmphora represents the smallest unit of gaz or 1/1,000,000,000th of an Amphora.
+
+### Roll
+
+Roll is a specific token used for staking on the Massa blockchain. A Roll token is required to create and endorse a block on the blockchain. The probability of being drawn to create or endorse a block is correlated to the number of rolls possed.
+Roll is expressed as an unsigned 64-bit integer.
+
+To simplify usage, the following standard units are recommended:
+
+kiloRoll (kROLL): 1 kiloRoll represents 1,000 rolls.
+megaRoll (MROLL): 1 megaRoll represents 1,000 kiloRolls or 1,000,000 rolls.
+
+### Transaction Fee
+
+A transaction fee is a small amount of Massa coin paid by the sender of a transaction to compensate the nodes in the Massa network for validating and executing the transaction.
+
+Transaction fees are denominated in Massa coin and are paid by the sender in addition to the amount being transacted. The fee amount is determined by the sender and is included in the transaction data.
+
+Note: Developers should use the same units implementation as for Massa coin values when handling transaction fee values.
+
+## Implementation
```

Comment:
In that case, I believe that the term 'usage' may be more appropriate.For me implementation suggests a more technical aspect like implementing a feature from a spec. Just a suggestion

Reply:
For the rest I think this is a well-written and understandable.

Reply:
Updated. I'll let you have a look and close this discussion if you think it's ok